### ISSUE TITLE
CodeGPTPlus/deepseek-coder-1.3b-typescript · Hugging Face

### ISSUE THREAD

**Issue by irthomasthomas** [irthomasthomas/undecidability#498]
- [ ] [CodeGPTPlus/deepseek-coder-1.3b-typescript · Hugging Face](https://huggingface.co/CodeGPTPlus/deepseek-coder-1.3b-typescript)

 # CodeGPTPlus/deepseek-coder-1.3b-typescript

This is a fine-tuned model by the CodeGPT team, specifically crafted for generating expert code in TypeScript. It is fine-tuned from `deepseek-ai/deepseek-coder-1.3b-base` with a dataset of 0.5B tokens, making it an excellent choice for precise and efficient TypeScript code generation.

The model uses a 16K window size and an additional fill-in-the-middle task for project-level code completion.

## How to Use

This model is for completion purposes only. Here are some examples of how to use the model:

### Running the model on a GPU

```python
from transformers import AutoTokenizer, AutoModelForCausalLM
tokenizer = AutoTokenizer.from_pretrained("CodeGPTPlus/deepseek-coder-1.3b-typescript", trust_remote_code=True)
model = AutoModelForCausalLM.from_pretrained("CodeGPTPlus/deepseek-coder-1.3b-typescript", trust_remote_code=True).cuda()

input_text = """<|fim begin|>function quickSort(arr: number[]): number[] {
  if (arr.length <= 1) {
    return arr;
  }
  const pivot = arr[0];
  const left = [];
  const right = [];
<|fim hole|>
  return [...quickSort(left), pivot, ...quickSort(right)];
}<|fim end|>"""

inputs = tokenizer(input_text, return_tensors="pt").to(model.device)
outputs = model.generate(**inputs, max_length=256)
print(tokenizer.decode(outputs[0], skip_special_tokens=True))
```

### Running with Ollama

- Model: [https://ollama.ai/codegpt/deepseek-coder-1.3b-typescript](https://ollama.ai/codegpt/deepseek-coder-1.3b-typescript)
- Command: `ollama run codegpt/deepseek-coder-1.3b-typescript`

### Running with Ollama and CodeGPT Autocomplete in VSCode

- Documentation: [https://docs.codegpt.co/docs/tutorial-features/code\_autocompletion](https://docs.codegpt.co/docs/tutorial-features/code_autocompletion)
- Select "Ollama - codegpt/deepseek-coder-1.3b-typescript" in the autocomplete model selector.

### Fill In the Middle (FIM)

```python
<|fim begin|>function quickSort(arr: number[]): number[] {
  if (arr.length <= 1) {
    return arr;
  }
  const pivot = arr[0];
  const left = [];
  const right = [];
<|fim hole|>
  return [...quickSort(left), pivot, ...quickSort(right)];
}<|fim end|>
```

### Training Procedure

The model was trained using the following hyperparameters:

- learning\_rate: 2e-05
- train\_batch\_size: 20
- eval\_batch\_size: 20
- seed: 42
- gradient\_accumulation\_steps: 2
- total\_train\_batch\_size: 40
- optimizer: Adam with betas=(0.9,0.999) and epsilon=1e-06
- lr\_scheduler\_type: cosine
- lr\_scheduler\_warmup\_steps: 261
- num\_epochs: 1

For more information, visit the [model page](https://huggingface.co/CodeGPTPlus/deepseek-coder-1.3b-typescript).

#### {   "label-name": "TypeScript-Code-Generation",   "description": "Model for generating TypeScript code",   "repo": "CodeGPTPlus/deepseek-coder-1.3b-typescript",   "confidence": 70.59 }
